Makefile.toml

```diff
@@ -330,6 +330,28 @@ done
 '''
 ]
 
+# A top level target for devs to ensure review and patch readiness
+[tasks.check]
+dependencies = ["check-build-links", "check-cargo-version", "check-fmt", "check-lints"]
+
+# Creates buildtime links if they don't exist yet
+[tasks.check-build-links]
+script = [
+'''
+if [ ! -f "${BUILDSYS_SOURCES_DIR}/models/src/variant/mod.rs" ]; then
+    pushd "${BUILDSYS_SOURCES_DIR}/models/src/variant" > /dev/null
+    ln -s ../variant_mod.rs mod.rs
+    popd > /dev/null
+fi
+
+if [ ! -d "${BUILDSYS_SOURCES_DIR}/models/src/variant/current" ]; then
+    pushd "${BUILDSYS_SOURCES_DIR}/models/src/variant" > /dev/null
+    ln -s ../aws-k8s-1.23 current
+    popd > /dev/null
+fi
+'''
+]
```

Comment:
This makes me uncomfortable, on account of [all](https://github.com/bottlerocket-os/bottlerocket/blob/e354d2cc7c907f717d8851aede27cd1f3991c001/Dockerfile#L36) [the](https://github.com/bottlerocket-os/bottlerocket/blob/e354d2cc7c907f717d8851aede27cd1f3991c001/Dockerfile#L114) [logic](https://github.com/bottlerocket-os/bottlerocket/blob/e354d2cc7c907f717d8851aede27cd1f3991c001/sources/models/build.rs#L54) involved in getting the right thing to happen. It'd be easy to break builds outside the container without realizing it, in a subtle way like reusing the model from a previous variant's build.

I'd recommend doing what `unit-tests` does instead and just export the expected variable in each of the tasks:
```
export VARIANT="${BUILDSYS_VARIANT}"
```

Reply:
The symlinks are created as part of `models`'s build script. Cargo Build scripts are only invoked upon compilation of the crate. Since `cargo fmt`, unlike `cargo test`, doesn't invoke `rustc`, the `build.rs` script won't get invoked.

Reply:
😿 

Maybe add `unit-tests` into the list before `check-fmt`, so the links get created first?

Ideally we'd be able to run clippy and `check-lints` could come before `check-fmt`, but that's probably too much to bite off right now.

Even if a workaround is required, I'd really prefer not to duplicate this sort of link logic outside the build script for `models` - the behavior is already very complicated.

Reply:
> Maybe add unit-tests into the list before check-fmt, so the links get created first?

I think that's the right thing here given the complexity of how those files are to be populated. Additionally, running the tests as part of a top level "check" isn't a bad idea anyways